tailcfg/tailcfg.go

```diff
@@ -2031,11 +2032,26 @@ type MapResponse struct {
 	// known problems). A non-zero length slice are the list of problems that
 	// the control plane sees.
 	//
+	// Either this will be set, or DisplayMessages will be set, but not both.
+	//
 	// Note that this package's type, due its use of a slice and omitempty, is
 	// unable to marshal a zero-length non-nil slice. The control server needs
 	// to marshal this type using a separate type. See MapResponse docs.
 	Health []string `json:",omitempty"`
 
+	// DisplayMessages, if non-nil, sets the health state of the node from the
+	// control plane's perspective. A nil value means no change from the
+	// previous MapResponse. A non-nil 0-length slice restores the health to
+	// good (no known problems). A non-zero length slice are the list of
+	// problems that the control plane sees.
+	//
+	// Either this will be set, or Health will be set, but not both.
+	//
+	// Note that this package's type, due its use of a slice and omitempty, is
+	// unable to marshal a zero-length non-nil slice. The control server needs
+	// to marshal this type using a separate type. See MapResponse docs.
+	DisplayMessages map[DisplayMessageID]DisplayMessage `json:",omitempty"`
```

Comment:
Since this is a map, it would be good to use the same API as for `PacketFilters` (keys are set to update or delete named messages, and a special key of `*` is used to replace the whole map)